x-pack/plugins/lens/public/visualizations/xy/visualization.tsx

```diff
@@ -126,6 +131,7 @@ export const getXyVisualization = ({
           seriesType: firstUsedSeriesType || state.preferredSeriesType,
           layerId,
           layerType,
+          indexPatternId: indexPatternId ?? core.uiSettings.get('defaultIndex'),
```

Comment:
Seems like these bits are related to query based annotations. Does it make sense to keep them here? We definitely need to extract/inject these references and as it's not happening on this PRs it's probably better to split it out of this PR.